src/catalog/src/durable/objects.rs

```diff
@@ -984,6 +986,26 @@ impl Snapshot {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Arbitrary)]
+pub struct FenceToken {
+    pub(crate) deploy_generation: u64,
+    pub(crate) epoch: Epoch,
+}
+
+impl PartialOrd for FenceToken {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for FenceToken {
```

Comment:
If the sort order is important it might be nice to add a small regression test

Reply:
I will add this in a followup PR.